purchase_ext_sst/models/purchase_order.py

```diff
@@ -67,3 +67,17 @@ def open_record(self):
             'context': {},
             'target': 'current',
         }
+
+    @api.multi
+    def write(self, vals):
+        res = super(PurchaseOrder, self).write(vals)
+        for purchase_order in self:
+            for order_line in purchase_order.order_line:
+                product = order_line.product_id.product_tmpl_id
+                if purchase_order.shop_id and product.shop_id != \
+                        purchase_order.shop_id:
+                    product.shop_id = purchase_order.shop_id
+                if purchase_order.purchased_by_id and \
+                   product.purchased_by_id != purchase_order.purchased_by_id:
+                    product.purchased_by_id = purchase_order.purchased_by_id
```

Comment:
@TimLai125 Sorry that I'm late to comment, but what about the case where PO `shop_id`/`purchased_by_id` is changed from something to blank?  In that case shouldn't we also update the corresponding product field to blank?